src/main/java/trackitnus/logic/parser/ParserUtil.java

```diff
@@ -198,9 +199,26 @@ public static LocalDate parseDate(String date) throws ParseException {
         requireNonNull(date);
         String trimmedDate = date.trim();
         try {
-            return LocalDate.parse(trimmedDate, Task.FORMATTER);
+            return LocalDate.parse(trimmedDate, DATE_PATTERN);
         } catch (DateTimeParseException e) {
-            throw new ParseException(Task.DATE_MESSAGE_CONSTRAINTS);
+            throw new ParseException(Messages.DATE_MESSAGE_CONSTRAINTS);
+        }
+    }
+
+    /**
+     * Parses a VALID {@code String date} into a {@code LocalDate}.
+     * Leading and trailing whitespaces will be trimmed.
+     * Crash the program if the date cannot be parsed (it's the caller responsibility to make sure the date passed in
+     * is valid!). The function is written so that the caller doesn't need to handle the possible exception since
+     * it's expected that no exception will be thrown.
+     */
+    public static LocalDate parseValidDate(String date) {
+        requireNonNull(date);
+        String trimmedDate = date.trim();
+        try {
+            return LocalDate.parse(trimmedDate, DATE_PATTERN);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("an invalid date is passed into parseValidDate");
```

Comment:
Hmm Actually how is this different from the normal parseDate? And why not just use the same Messages.DATE_MESSAGE_CONSTRAINTS? In any case, I think the first letter should be capitalised: `An invalid date...`

Reply:
it is very different because the IllegalArgumentException is an unchecked exception, which doesn't require the caller to try/catch for it. In other words, this function will crash the program (assert-like) if the provided date is invalid!

Reply:
Also I will capitalize the "a" now

Reply:
Done!

Reply:
Sorry actually for grammar it should be "An invalid date has been passed into parseValidDate"

Reply:
done again!